openedx/features/enterprise_support/api.py

```diff
@@ -355,7 +354,7 @@ def consent_needed_for_course(request, user, course_id, enrollment_exists=False)
     if not enterprise_learner_details:
         consent_needed = False
     else:
-        client = ConsentApiClient()
+        client = ConsentApiClient(user=request.user)
```

Comment:
@haikuginger Same question here. Are we good with using the request user?

Reply:
Request user is fine here.